src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueController.java

```diff
@@ -191,16 +193,38 @@
         @RequestParam(value = "court-type-id", required = false)
         @ApiParam(name = "court-type-id",
             value = "Alphabets and Numeric values only allowed in comma separated format")
-            String courtTypeId) {
+            String courtTypeId,
+        @RequestParam(value = "is_hearing_location", required = false)
+        @ApiParam(name = "is_hearing_location",
+            value = "Allowed values are \"Y\" or \"N\"")
+            String isHearingLocation,
+        @RequestParam(value = "is_case_management_location", required = false)
+        @ApiParam(name = "is_case_management_location",
+            value = "Allowed values are \"Y\" or \"N\"")
+            String isCaseManagementLocation,
+        @RequestParam(value = "location_type", required = false)
+        @ApiParam(name = "location_type",
+            value = "allowed values are CTSC, NBC, Court,CCBC etc")
+            String locationType,
+        @RequestParam(value = "is_Temporary_Location", required = false)
+        @ApiParam(name = "is_Temporary_Location",
+            value = "Allowed values are \"Y\" or \"N\"")
+            String isTemporaryLocation
+    ) {
         String trimmedSearchString = searchString.strip();
         validateSearchString(trimmedSearchString);
         if (StringUtils.isNotBlank(courtTypeId)) {
             validateCourtTypeId(courtTypeId);
         }
+
+        var requestParam = new CourtVenueRequestParam();
```

Comment:
can be moved to util method along with validations also , which is actually done in service method

Reply:
here in this line we are only setting parameter to object and passing instead of individual param.
Validations is util iteself.